build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -118,7 +127,7 @@ public async Task<List<DotNetVersion>> GetNewDotNetVersionsAsync()
                 }
             }
 
-            return dotNetVersions.OrderBy(v => v.Version).ToList();
+            return dotNetVersions.OrderBy(v => new Version(v.Version)).ToList();
```

Comment:
In my opinion, it may be better if we make both `DotNetVersion` and `PythonVersion` implement the `IComparable` interface so that they both implement `CompareTo` and can inherently be sorted without needing to rely on a specific property or make these types of conversions each time we need to.

The change in the `DotNetVersion` class would be the following:

```cs
public class DotNetVersion : IComparable
{
    // ...

    public int CompareTo(object obj)
    {
        var objDotNetVersion = obj as DotNetVersion;
        if (objDotNetVersion == null)
        {
            return 1;
        }

        var thisVersion = new Version(this.Version);
        var objVersion = new Version(objDotNetVersion.Version);
        return thisVersion.CompareTo(objVersion);
   }
}
```

The above line in your PR would then just resolve to `return dotNetVersions.Order().ToList();`

Reply:
This makes sense, thanks!